app/controllers/concerns/overdue_list_filtering.rb

```diff
@@ -0,0 +1,39 @@
+module OverdueListFiltering
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :set_accessible_facilities, only: [:index]
+    before_action :populate_districts, only: [:index]
+    before_action :set_selected_district, only: [:index]
+    before_action :populate_facilities, only: [:index]
+    before_action :set_selected_facility, only: [:index]
+
+    private
+
+    def set_accessible_facilities
+      @accessible_facilities = current_admin.accessible_facilities(:manage_overdue_list)
+    end
+
+    def populate_districts
+      @districts =
+        Region.district_regions.where(source_id: @accessible_facilities.map(&:facility_group_id).uniq).order(:name)
```

Comment:
Can we traverse this via facility regions instead? I'd like to minimize reliance on FGs and maximize reliance on regions, so it becomes easier to modify/do away with FGs in the future.

Reply:
Addressed in 7f46c0a12